contracts/ERC721MLite.sol

```diff
@@ -0,0 +1,568 @@
+        _currentBaseURI = baseURI;
+        emit SetBaseURI(baseURI);
+    }
+
+    /**
+     * @dev Returns token URI suffix.
+     */
+    function getTokenURISuffix()
+        external
+        view
+        override
+        returns (string memory)
+    {
+        return _tokenURISuffix;
+    }
+
+    /**
+     * @dev Sets token URI suffix. e.g. ".json".
+     */
+    function setTokenURISuffix(string calldata suffix) external onlyOwner {
+        _tokenURISuffix = suffix;
+    }
+
+    /**
+     * @dev Returns token URI for a given token id.
+     */
+    function tokenURI(uint256 tokenId)
+        public
+        view
+        override(ERC721A, IERC721A)
+        returns (string memory)
+    {
+        if (!_exists(tokenId)) revert URIQueryForNonexistentToken();
+
+        string memory baseURI = _currentBaseURI;
+        return
+            bytes(baseURI).length != 0
+                ? string(
+                    abi.encodePacked(
+                        baseURI,
+                        _toString(tokenId),
+                        _tokenURISuffix
+                    )
+                )
+                : "";
+    }
+
+    /**
+     * @dev Returns data hash for the given minter, qty and timestamp.
+     */
+    function _getCosignDigest(
+        address minter,
+        uint32 qty,
+        uint64 timestamp
+    ) public view returns (bytes32) {
+        if (_cosigner == address(0)) revert CosignerNotSet();
+        return
+            keccak256(
+                abi.encodePacked(
+                    address(this),
+                    minter,
+                    qty,
+                    _cosigner,
+                    timestamp,
+                    _chainID(),
+                    getCosignNonce(minter)
+                )
+            ).toEthSignedMessageHash();
+    }
+
+    /**
+     * @dev Validates the the given signature.
+     */
+    function assertValidCosign(
+        address minter,
+        uint32 qty,
+        uint64 timestamp,
+        bytes memory signature
+    ) public view {
+        if (
+            !SignatureChecker.isValidSignatureNow(
+                _cosigner,
+                _getCosignDigest(minter, qty, timestamp),
+                signature
+            )
+        ) revert InvalidCosignSignature();
+    }
+
+    /**
+     * @dev Returns the current active stage based on timestamp.
+     */
+    function getActiveStageFromTimestamp(uint64 timestamp)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        for (uint256 i = 0; i < _mintStages.length; i++) {
+            if (
+                timestamp >= _mintStages[i].startTimeUnixSeconds &&
+                timestamp < _mintStages[i].endTimeUnixSeconds
+            ) {
+                return i;
+            }
+        }
+        revert InvalidStage();
+    }
+
+    /**
+     * @dev Returns mint currency address.
+     */
+    function getMintCurrency() external pure override returns (address) {
```

Comment:
Could we remove this? We don't call it anywhere in the frontend

Reply:
hmmm, you added this function into the interface? Again we can clean them in a separate PR.

Reply:
Oh yeah I did - sounds good removing this and the crossmint getter from the interface in another PR